src/main/scala/com/fulcrumgenomics/bam/EstimatePoolingFractions.scala

```diff
@@ -0,0 +1,242 @@
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(vcf :: bam :: intervals.toList)
+
+  private val Ci99Width = 2.58 // Width of a 99% confidence interval in units of std err
+
+  /* Class to hold information about a single locus. */
+  case class Locus(chrom: String, pos: Int, ref: Char, alt: Char, expectedSampleFractions: Array[Double], var observedFraction: Option[Double] = None)
+
+  override def execute(): Unit = {
+    val vcfReader = new VCFFileReader(vcf.toFile)
+    val sampleNames = pickSamplesToUse(vcfReader)
+    val intervals   = loadIntervals
+
+    // Get the expected fractions from the VCF
+    val vcfIterator = constructVcfIterator(vcfReader, intervals, sampleNames)
+    val loci = vcfIterator.filterNot(v => this.nonAutosomes.contains(v.getContig)).map { v => Locus(
+      chrom = v.getContig,
+      pos = v.getStart,
+      ref = v.getReference.getBaseString.charAt(0),
+      alt = v.getAlternateAllele(0).getBaseString.charAt(0),
+      expectedSampleFractions = sampleNames.map { s => val gt = v.getGenotype(s); if (gt.isHomRef) 0 else if (gt.isHet) 0.5 else 1.0 }
+    )}.toArray
+
+    logger.info(s"Loaded ${loci.length} bi-allelic SNPs from VCF.")
+
+    val coveredLoci = fillObserveredFractionAndFilter(loci, this.minCoveragePerSample * sampleNames.length)
+
+    logger.info(s"Regressing on ${coveredLoci.length} of ${loci.length} that met coverage requirements.")
+    val regression = new OLSMultipleLinearRegression
+    regression.setNoIntercept(true) // Intercept should be at 0!
+    regression.newSampleData(
+      coveredLoci.map(_.observedFraction.getOrElse(unreachable("observed fraction must be defined"))),
+      coveredLoci.map(_.expectedSampleFractions)
+    )
+
+    val regressionParams = regression.estimateRegressionParameters()
+    val total            = regressionParams.sum
+    val fractions        = regressionParams.map(_ / total)
+    val stderrs          = regression.estimateRegressionParametersStandardErrors().map(_ / total)
+    logger.info(s"R^2 = ${regression.calculateRSquared()}")
+    logger.info(s"Sum of regression parameters = ${total}")
+
+    val metrics = sampleNames.toSeq.zipWithIndex.map { case (sample, index) =>
+      val sites      = coveredLoci.count(l => l.expectedSampleFractions(index) > 0)
+      val singletons = coveredLoci.count(l => l.expectedSampleFractions(index) > 0 && l.expectedSampleFractions.sum == l.expectedSampleFractions(index))
+      new PoolingFractionMetric(
+        sample             = sample,
+        variant_sites      = sites,
+        singletons         = singletons,
+        estimated_fraction = fractions(index),
+        standard_error     = stderrs(index),
+        ci99_low           = max(0, fractions(index) - stderrs(index)*Ci99Width),
+        ci99_high          = min(1, fractions(index) + stderrs(index)*Ci99Width))
+    }
+
+    Metric.write(metrics, path=output)
+
+    if (regression.estimateRegressionParameters().exists(_ < 0)) {
```

Comment:
This is actually testing for a different condition that is somewhat hard to induce via a test.  It is testing to see if one or more of the samples is estimated to be a negative fraction of the pool, which _can_ but won't necessary happen if you list one or more samples that aren't actually in the pool